src/backends/backend/triton_model.cc

```diff
@@ -564,6 +564,67 @@ TRITONBACKEND_RequestRelease(
   return nullptr;  // success
 }
 
+///
+/// TRITONBACKEND_State
+///
+
+TRITONSERVER_Error*
+TRITONBACKEND_StateUpdate(TRITONBACKEND_State* state)
+{
+  InferenceRequest::State* ts =
+      reinterpret_cast<InferenceRequest::State*>(state);
+  auto status = ts->Update();
+
+  if (!status.IsOk()) {
+    return TRITONSERVER_ErrorNew(
+        StatusCodeToTritonCode(status.StatusCode()), status.Message().c_str());
+  }
+
+  return nullptr;  // success
+}
+
+TRITONSERVER_Error*
+TRITONBACKEND_StateNew(
+    TRITONBACKEND_State** state, TRITONBACKEND_Request* request,
+    const char* name, const TRITONSERVER_DataType datatype,
+    const int64_t* shape, const uint32_t dims_count)
+{
+  InferenceRequest* tr = reinterpret_cast<InferenceRequest*>(request);
+  std::vector<int64_t> lshape(shape, shape + dims_count);
+  InferenceRequest::State* lstate;
+  Status status =
+      tr->AddState(name, TritonToDataType(datatype), lshape, &lstate);
+  *lstate->MutableShape() = lshape;
```

Comment:
Why do you need to call this, doesn't AddState set the shape? In any case it should be after the status check.

Reply:
I'll modify this. Because State is based on input. The shape will be added to `original_shape_` instead of `shapes_` attribute.